go-controller/pkg/ovn/controller/apbroute/external_controller_pod_test.go

```diff
@@ -178,6 +178,32 @@ var _ = Describe("OVN External Gateway pod", func() {
 			eventuallyExpectNumberOfPolicies(1)
 			eventuallyExpectConfig(noMatchPolicy.Name, expectedPolicy, expectedRefs)
 		})
+		It("processes the pod that is in Pending phase and then changes to Running with an assigned IP", func() {
+
+			targetPodPending := newPodWithPhaseAndIP("pod_pending", namespaceTarget2.Name, corev1.PodPending, "",
+				map[string]string{"key": "pod", "name": "pod_pending"})
+			namespaceTargetWithPendingPod := newNamespaceWithPods(namespaceTarget2.Name, targetPodPending)
+
+			initController([]runtime.Object{namespaceGW, namespaceTarget2, pod1, targetPodPending}, []runtime.Object{dynamicPolicy})
+
+			expectedPolicy, expectedRefs := expectedPolicyStateAndRefsWithFailedToApply(
```

Comment:
I guess it won't have failed status anymore?

Reply:
yep, I will revert this change since it's not needed now 😄